src/AeroSharp/MapBuilder.cs

```diff
@@ -0,0 +1,159 @@
+﻿using AeroSharp.Connection;
+using AeroSharp.DataAccess;
+using AeroSharp.DataAccess.Configuration;
+using AeroSharp.DataAccess.Internal;
+using AeroSharp.DataAccess.MapAccess;
+using AeroSharp.DataAccess.MapAccess.Generators;
+using AeroSharp.DataAccess.MapAccess.Parsers;
+using AeroSharp.DataAccess.Validation;
+using AeroSharp.Serialization;
+using FluentValidation;
+using System;
+
+namespace AeroSharp;
+
+/// <inheritdoc cref="IMapBuilder"/>
+public sealed class MapBuilder : IDataContextBuilder<IMapBuilder>, IMapBuilder
+{
+    private readonly IClientProvider _clientProvider;
+
+    private readonly IMapBinParser _mapBinParser;
+
+    private readonly AbstractValidator<DataContext> _dataContextValidator;
+
+    private readonly AbstractValidator<MapContext> _mapContextValidator;
+
+    private readonly AbstractValidator<WriteConfiguration> _writeConfigurationValidator;
+
+    private DataContext _dataContext;
+
+    private ISerializer _serializer;
+
+    private WriteConfiguration _writeConfiguration;
+
+    private MapConfiguration _mapConfiguration;
+
+    internal MapBuilder(IClientProvider clientProvider)
+    {
+        _clientProvider = clientProvider;
+
```

Comment:
Nit: is the blank line here intentional?

Reply:
It is intentional, since it visually separates the fields we're setting from input args versus what we're just new-ing up in the constructor. I think it improves readability, but can remove it if that's just me. 😅